ygen/protogen.go

```diff
@@ -198,7 +219,21 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 	}
 	sort.Strings(fNames)
 
+	// If the message that we are generating for is a list, then we explicitly
+	// want to skip it keys to ensure that they are not duplicated.
+	skipFields := map[string]bool{}
+	if msg.entry.IsList() && msg.entry.Key != "" {
```

Comment:
Actually this check isn't really required, since key == "" in all cases that this isn't a keyed list. I kept the check because it seems to add to robustness that something unexpected isn't there.

Reply:
ACK